nipype/algorithms/confounds.py

```diff
@@ -1193,9 +1194,12 @@ def compute_noise_components(imgseries, mask_images, num_components,
         try:
             u, _, _ = np.linalg.svd(M, full_matrices=False)
```

Comment:
Nested `try`/`except` blocks get a little hard to reason about. How about a helper function that will do the fall-back:

```Python
# Use the numpy.linalg.svd protocol
def fallback_svd(a, full_matrices=True, compute_uv=True):
    try:
        return np.linalg.svd(a, full_matrices=full_matrices, compute_uv=compute_uv)
    except np.linalg.LinAlgError:
        pass

    from scipy.linalg import svd
    return svd(a, full_matrices=full_matrices, compute_uv=compute_uv, lapack_driver='gesvd')
```

And then we just replace the above with:
```suggestion
            u, _, _ = fallback_svd(M, full_matrices=False)
```

Reply:
And to be clear, I'm suggesting we place the above `fallback_svd` function at the file level.

Reply:
Great idea! By putting it at the file level do you mean putting it at the top of the file?

Reply:
Yes. Anywhere, really, but not nested within a class or another function. The top is fine.